### PR TITLE
treemix: init at 1.13

### PR DESCRIPTION
**Pull request by bzizou**
###### Things done

Application needed by our biologists in our HPC center

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by smancill — DISMISSED** [review]

File: pkgs/applications/science/biology/treemix/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    url = "https://bitbucket.org/nygcresearch/treemix/downloads/${pname}-${version}.tar.gz";
+    sha256 = "1nd3rzsdgk47r8b8k43mdfvaagln533sm08s1jr0dz8km8nlym7y";
+  };
+
+  buildInputs = [ zlib gsl boost ];
+
+  meta = with lib ; {
+    description = "Inference of patterns of population splitting and mixing from genome-wide allele frequency data";
+    homepage = "https://bitbucket.org/nygcresearch/treemix/wiki/Home";
+    license = licenses.gpl3;
```
Comment (applePrincess) [inline pkgs/applications/science/biology/treemix/default.nix:23]: gpl3 is deprecated it should be `licenses.gpl3Only`

**Review by applePrincess — APPROVED** [review]

File: pkgs/applications/science/biology/treemix/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+, gsl
+, boost
+}:
+
+stdenv.mkDerivation rec {
+  pname = "treemix";
+  version = "1.13";
+
+  src = fetchurl {
+    url = "https://bitbucket.org/nygcresearch/treemix/downloads/${pname}-${version}.tar.gz";
```
Comment (Mindavi) [inline pkgs/applications/science/biology/treemix/default.nix:14]: I guess fetchBitbucket is not a thing :)

File: pkgs/applications/science/biology/treemix/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  buildInputs = [ zlib gsl boost ];
+
+  meta = with lib ; {
+    description = "Inference of patterns of population splitting and mixing from genome-wide allele frequency data";
+    homepage = "https://bitbucket.org/nygcresearch/treemix/wiki/Home";
+    license = licenses.gpl3Only;
+    maintainers = [ maintainers.bzizou ];
+    platforms = platforms.all;
```
Comment (Mindavi) [inline pkgs/applications/science/biology/treemix/default.nix:25]: Pretty sure this needs to be either `platforms.linux` or `platforms.unix`

**Review by Mindavi — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/biology/treemix/default.nix (hunk 3 above)
Comment (smancill) [reply pkgs/applications/science/biology/treemix/default.nix:25]: @bzizou please resolve this.

File: pkgs/applications/science/biology/treemix/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/treemix/default.nix:28]:
```suggestion
  };
}
```

**Pull request merged**